src/AppRun.c

```diff
@@ -25,182 +25,298 @@ THE SOFTWARE.
+  * \param the file path to check
+  * \return true if the path points to a .desktop file, else false.
+  */
+ int filter(const struct dirent* dir) {
+     // Get pointer to file path.
+     char* p = (char*) &dir->d_name;
+     // Read only the file extension (start reading from last dot in path)
+     p = strrchr(p, '.');
+     // Return true if the file extension .desktop is found.
+     return (p && !strcmp(p, ".desktop")) ? 1 : 0;
+ }
+
 int main(int argc, char *argv[]) {
-    char *appdir = dirname(realpath("/proc/self/exe", NULL));
-    if (!appdir)
+    // Get path to the application directory.
+    char* appdir = dirname(realpath("/proc/self/exe", NULL));
+    /* If there's a problem accessing the application directory,
+     * throw fatal error.
+     */
+    if (!appdir) {
         die("Could not access /proc/self/exe\n");
+    }
 
+    // Check for .desktop files
+    // The return code for whether we found a .desktop file
     int ret;
-    struct dirent **namelist;
+    // An array of pointers to lists of file paths.
+    struct dirent** namelist;
+    // Scan for .desktop files.
     ret = scandir(appdir, &namelist, filter, NULL);
 
     if (ret == 0) {
-        die("No .desktop files found\n");
+        // If we found no .desktop files, there's nothing to run. Fatal error.
+        die("No .desktop files found.\n");
     } else if(ret == -1) {
+        // If there was an error with scandir(), we can't go on. Fatal error.
         die("Could not scan directory %s\n", appdir);
     }
 
-    /* Extract executable from .desktop file */
-    char *desktop_file = calloc(LINE_SIZE, sizeof(char));
+    /* Extract executable from the first (and presumably only) desktop
+     * file found in the application directory.
+     */
+    char* desktop_file = calloc(LINE_SIZE, sizeof(char));
     snprintf(desktop_file, LINE_SIZE, "%s/%s", appdir, namelist[0]->d_name);
-    FILE *f     = fopen(desktop_file, "r");
-    char *line  = malloc(LINE_SIZE);
-    char *exe   = line+5;
-    size_t n    = LINE_SIZE;
-
+    // Open the desktop file for reading.
+    FILE* f = fopen(desktop_file, "r";)
+    // Store the line we're reading.
+    char* line = malloc(LINE_SIZE);
+    // Skip the `Exec=` prefix on `line`, only looking at the stored value.
+    char* exe = line + 5;
+    size_t n = LINE_SIZE;
+
+    /* Search each line of our file for the `Exec=` entry.
+     * To search, we store the line in `line`, and then check for the
+     * correct prefix. If it doesn't match, we keep looking.
+     */
     do {
-        if (getline(&line, &n, f) == -1)
-            die("Executable not found, make sure there is a line starting with 'Exec='\n");
-    } while(strncmp(line, "Exec=", 5));
+        // If we reach the end of the file without finding `Exec=`, fatal error.
+        if (getline(&line, &n, f) == -1) {
+            die("Executable not found. Make sure there is a line starting with 'Exec='\n");
+        }
+    } while(strncmp(line, "Exec", 5));
```

Comment:
You removed the `=` sign, the `strncmp` call cannot work any more (comparing 5 chars, but `Exec` is only 4 chars long, so this is basically `strcmp`; we want to check whether the string contains the prefix, though).